Core/include/Acts/Fitter/KalmanFitter.hpp

```
@@ -981,8 +981,9 @@ class KalmanFitter {
     ACTS_VERBOSE("Preparing " << sourcelinks.size() << " input measurements");
     std::map<const Surface*, source_link_t> inputMeasurements;
     for (const auto& sl : sourcelinks) {
-      const Surface* srf = &sl.referenceSurface();
-      inputMeasurements.emplace(srf, sl);
+      const Surface* srf = sl.referenceSurface();
+      if(srf != nullptr)
```

I don’t think this should be a silent failure.